commcare_connect/program/helpers.py

```diff
@@ -0,0 +1,56 @@
+from django.db.models import Avg, Count, DurationField, ExpressionWrapper, F, OuterRef, Q, Subquery
+
+from commcare_connect.opportunity.models import UserVisit, VisitValidationStatus
+from commcare_connect.program.models import ManagedOpportunity, Program
+
+
+def get_annotated_managed_opportunity(program: Program):
+    filter_for_valid__visit_date = ~Q(
+        opportunityaccess__uservisit__status__in=[
+            VisitValidationStatus.over_limit,
+            VisitValidationStatus.trial,
+        ]
+    )
+
+    earliest_visits = (
+        UserVisit.objects.filter(
+            opportunity_access=OuterRef("opportunityaccess"),
+        )
+        .exclude(status__in=[VisitValidationStatus.over_limit, VisitValidationStatus.trial])
+        .order_by("visit_date")
+        .values("visit_date")[:1]
```

Comment:
nit: You can use `.first()` here instead of the slice

Reply:
I tried .first() but subquery here expects a queryset.